py/tests/integration/test_spend_tx.py

```diff
@@ -88,6 +88,83 @@ def test_successful():
     common.stop_node(alice_node)
     shutil.rmtree(root_dir)
 
+def test_not_enough_tokens():
+    # Bob should not be able to send more tokens than he has
+    #
+    # Let's say Bob has 100 tokens. He should not be able to send more than
+    # 100 tokens to Alice.
+    #
+    # If there's an incoming but unconfirmed deposit into Bob's account then Bob
+    # should not be able to use the incoming tokens until the spend transaction
+    # they are in is confirmed.
+    test_settings = settings["test_not_enough_tokens"]
+    coinbase_reward = common.coinbase_reward() 
+    (root_dir, bob_node, bob_api, bob_top) = setup_node_with_tokens(test_settings, "bob") 
+
+    spend_tx_amt = test_settings["spend_tx"]["amount"]
+    spend_tx_fee = test_settings["spend_tx"]["fee"]
+    bob_balance = bob_api.get_account_balance()
+    bob_has_not_enough_tokens = bob_balance.balance < spend_tx_amt + spend_tx_fee
+    assert_equals(bob_has_not_enough_tokens, True)
+    print("Bob initial balance is " + str(bob_balance.balance) +
+            " and he will try to spend " + str(spend_tx_amt + spend_tx_fee))
+
+    alice_balance0 = {"balance":0} 
+    try:
+        alice_balance0 = bob_api.get_account_balance(pub_key=test_settings["spend_tx"]["alice_pubkey"])
+    except ApiException as e:
+        assert_equals(e.status, 404) # Alice has no account yet
```

Comment:
Same as https://github.com/aeternity/epoch/pull/383/files#r152543959

Not needed for this PR though.

Reply:
same comment as above